site/zenodo_rdm/migrator/load.py

```diff
@@ -0,0 +1,93 @@
+
+"""Zenodo migrator files loader."""
+
+import contextlib
+from pathlib import Path
+
+import psycopg
+from invenio_rdm_migrator.load.postgresql import _ts
+from invenio_rdm_migrator.streams.files.models import (
+    FilesBucket,
+    FilesInstance,
+    FilesObjectVersion,
+)
+
+
+class ZenodoFilesLoad:
+    """Zenodo to RDM Files class for data loading."""
+
+    def __init__(self, db_uri, tmp_dir, **kwargs):
+        """Constructor."""
+        self.db_uri = db_uri
+        self.tmp_dir = Path(tmp_dir)
+        self.models = [FilesInstance, FilesBucket, FilesObjectVersion]
+
+    def _load(self):
+        """Bulk load CSV table files.
+
+        Loads the tables in the order given by the generator.
+        """
+
+        def get_cols(model):
+            from dataclasses import fields
+
+            return ", ".join([f.name for f in fields(model)])
+
+        with psycopg.connect(self.db_uri) as conn:
+            for model in self.models:
+                name = model._table_name
+                fpath = Path(self.tmp_dir) / f"{name}.csv"
+                cols = get_cols(model)
+                if fpath.exists():
+                    # total file size for progress logging
+                    file_size = fpath.stat().st_size
+
+                    print(f"[{_ts()}] COPY FROM {fpath}")  # TODO: logging
+                    with contextlib.ExitStack() as stack:
+                        cur = stack.enter_context(conn.cursor())
+                        copy = stack.enter_context(
+                            cur.copy(f"COPY {name} ({cols}) FROM STDIN (FORMAT csv)")
+                        )
+                        fp = stack.enter_context(open(fpath, "r"))
+
+                        block_size = 8192
+
+                        def _data_blocks(block_size=8192):
+                            data = fp.read(block_size)
+                            while data:
+                                yield data
+                                data = fp.read(block_size)
+
+                        for idx, block in enumerate(_data_blocks(block_size)):
+                            if idx % 100:
+                                cur_bytes = idx * block_size
+                                percentage = (cur_bytes / file_size) * 100
+                                progress = (
+                                    f"{cur_bytes}/{file_size} ({percentage:.2f}%)"
+                                )
+                                print(f"[{_ts()}] {name}: {progress}")
+                            copy.write(block)
+                else:
+                    # FIXME: log a WARNING/ERROR
+                    print(f"[{_ts()}] {name}: no data to load from {fpath}")
+                conn.commit()
+
+    def run(self, entries, cleanup=False):
+        """Load entries."""
+        from datetime import datetime
+
+        start_time = datetime.now()
+        print(f"Stream started {start_time.isoformat()}")
+
+        self._load()
```

Comment:
Should call `post_load` afterwards (i.e. super()) and reuse form parent base Load class.